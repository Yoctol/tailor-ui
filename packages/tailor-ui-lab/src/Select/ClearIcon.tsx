import React, { FC } from 'react';
import styled from 'styled-components';
import { MdClose } from 'react-icons/md';

import { Icon } from 'tailor-ui';

import { getDataTestId } from './utils';

interface ClearIconProps {
  clearSelection: () => void;
  'data-testid'?: string;
}

const ClearIconWrapper = styled.div`
  display: inline-flex;
  flex: none;

  &:hover svg {
    fill: ${p => p.theme.colors.gray700} !important;
  }
`;

const ClearIcon: FC<ClearIconProps> = ({ clearSelection, ...props }) => (
  <ClearIconWrapper>
    <Icon
      type={MdClose}
      fill="gray400"
      size="20"
      cursor="pointer"
      onClick={e => {
        e.stopPropagation();
        clearSelection();
      }}
      {...getDataTestId(props, 'clear-icon')}
    />
  </ClearIconWrapper>
);

export default ClearIcon;
