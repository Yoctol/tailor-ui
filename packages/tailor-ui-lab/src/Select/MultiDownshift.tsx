import Downshift, {
  ControllerStateAndHelpers,
  DownshiftInterface,
} from 'downshift';
import React, { Component, ReactNode } from 'react';

import { itemToString } from './utils';

type Option =
  | {
      label: string;
      value: string;
    }
  | string
  | number;

type MultiDownshiftProps = DownshiftInterface<Option> & {
  selectedItem: Option[];
  onSelect?: (
    options: Option[],
    stateAndHelpers: ControllerStateAndHelpers<Option>
  ) => void;
  onChange?: (
    options: Option[],
    stateAndHelpers: ControllerStateAndHelpers<Option>
  ) => void;
  children: (options: ControllerStateAndHelpers<Option>) => ReactNode;
};

interface MultiDownshiftState {
  selectedItems: Option[];
}

class MultiDownshift extends Component<
  MultiDownshiftProps,
  MultiDownshiftState
> {
  state: MultiDownshiftState = {
    selectedItems: this.props.selectedItem || [],
  };

  stateReducer = (state: any, changes: any) => {
    switch (changes.type) {
      case Downshift.stateChangeTypes.blurButton:
        return {};
      case Downshift.stateChangeTypes.keyDownEnter:
      case Downshift.stateChangeTypes.mouseUp:
      case Downshift.stateChangeTypes.clickItem:
        return {
          ...changes,
          highlightedIndex: state.highlightedIndex,
          isOpen: true,
          inputValue: '',
        };
      default:
        return changes;
    }
  };

  handleSelection = (selectedItem: Option, downshift: any) => {
    const callOnChange = () => {
      const { onSelect, onChange } = this.props;
      const { selectedItems } = this.state;
      if (onSelect) {
        onSelect(selectedItems, this.getStateAndHelpers(downshift));
      }
      if (onChange) {
        onChange(selectedItems, this.getStateAndHelpers(downshift));
      }
    };

    if (
      this.state.selectedItems
        .map(itemToString)
        .includes(itemToString(selectedItem))
    ) {
      this.removeItem(selectedItem, callOnChange);
    } else {
      this.addSelectedItem(selectedItem, callOnChange);
    }
  };

  removeItem = (item: Option, cb?: () => void) => {
    this.setState(({ selectedItems }) => {
      return {
        selectedItems: selectedItems.filter(
          i => itemToString(i) !== itemToString(item)
        ),
      };
    }, cb);
  };

  addSelectedItem(item: Option, cb: () => void) {
    this.setState(
      ({ selectedItems }) => ({
        selectedItems: [...selectedItems, item],
      }),
      cb
    );
  }

  getRemoveButtonProps = ({
    onClick,
    item,
    ...props
  }: {
    onClick?: (e: any) => void;
    item: Option;
  }) => {
    return {
      onClick: (e: any) => {
        // TODO: use something like downshift's composeEventHandlers utility instead
        if (onClick) {
          onClick(e);
        }
        e.stopPropagation();
        this.removeItem(item);
      },
      ...props,
    };
  };

  getStateAndHelpers(downshift: ControllerStateAndHelpers<any>) {
    const { selectedItems } = this.state;
    const { getRemoveButtonProps, removeItem } = this;
    return {
      getRemoveButtonProps,
      removeItem,
      selectedItems,
      ...downshift,
    };
  }

  render() {
    const { children, ...props } = this.props;
    // TODO: compose together props (rather than overwriting them) like downshift does
    return (
      <Downshift
        {...props}
        stateReducer={this.stateReducer}
        onChange={this.handleSelection}
        selectedItem={null}
      >
        {downshift => children(this.getStateAndHelpers(downshift))}
      </Downshift>
    );
  }
}

export default MultiDownshift;
