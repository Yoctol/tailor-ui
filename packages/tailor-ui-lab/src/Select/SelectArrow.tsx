import React, { FC } from 'react';
import { MdKeyboardArrowUp } from 'react-icons/md';
import { animated, config, useSpring } from 'react-spring';

import { Icon } from 'tailor-ui';

const SelectArrow: FC<{ on: boolean }> = ({ on }) => {
  const { deg } = useSpring({
    deg: on ? 0 : 180,
    config: config.stiff,
  });

  return (
    <animated.i
      style={{
        transform: deg.interpolate(d => `rotate(${d}deg)`),
        flex: 'none',
        pointerEvents: 'none',
        display: 'inline-flex',
      }}
    >
      <Icon type={MdKeyboardArrowUp} fill="gray400" />
    </animated.i>
  );
};

export default SelectArrow;
