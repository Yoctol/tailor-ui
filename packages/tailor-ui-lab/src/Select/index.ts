import { Option as _Option } from './SelectOptions';

export * from './Select';
export type Option = _Option;
