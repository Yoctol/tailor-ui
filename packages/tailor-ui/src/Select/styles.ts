import styled, { css, keyframes } from 'styled-components';

export const SelectWrapper = styled.div`
  display: inline-flex;
  cursor: pointer;
`;

export interface StyledSelectOptionProps {
  active: boolean;
  hovered: boolean;
  disabled?: boolean;
}

export const StyledSelectOption = styled.div<StyledSelectOptionProps>`
  display: flex;
  align-items: center;
  background-color: ${(p) =>
    p.hovered ? p.theme.colors.primaryLight : p.theme.colors.light};
  color: ${(p) => {
    if (p.hovered) {
      return p.theme.colors.light;
    }
    if (p.active) {
      return p.theme.colors.primaryLight;
    }
    if (p.disabled) {
      return p.theme.colors.gray300;
    }
    return p.theme.colors.gray700;
  }};
  font-size: ${(p) => p.theme.fontSizes.sm};
  cursor: ${(p) => (p.disabled ? 'not-allowed' : 'pointer')};

  > div {
    padding: 0 24px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;

export const StyledSelectedOption = styled.div<{ size: 'sm' | 'md' | 'lg' }>`
  display: inline-flex;
  align-items: center;
  max-width: 100%;
  margin-right: 8px;
  padding: 0 8px;
  border: ${(p) => p.theme.borders.base};
  border-radius: ${(p) => p.theme.radii.base};
  border-color: ${(p) => p.theme.colors.primaryLight};
  background-color: ${(p) => p.theme.colors.surface2};
  color: ${(p) => p.theme.colors.primaryLight};
  font-size: ${(p) => p.theme.fontSizes.sm};

  ${({ size, theme: { heights } }) =>
    ({
      sm: css`
        min-height: calc(${heights.sm} - 4px);
        margin-top: 0.5px;
        margin-bottom: 0.5px;
      `,
      md: css`
        min-height: calc(${heights.base} - 10px);
        margin-top: 2px;
        margin-bottom: 2px;
      `,
      lg: css`
        min-height: calc(${heights.lg} - 14px);
        margin-top: 3px;
        margin-bottom: 3px;
      `,
    }[size])};

  > span {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  svg {
    fill: ${(p) => p.theme.colors.primaryLight2};
  }

  &:hover svg {
    fill: ${(p) => p.theme.colors.primaryLight};
  }
`;

export interface StyledSelectProps {
  focused: boolean;
  disabled: boolean;
  invalid: boolean;
  multiple: boolean;
  size?: 'sm' | 'md' | 'lg';
}

export const StyledSelect = styled.div<StyledSelectProps>`
  display: inline-flex;
  align-items: center;
  width: 100%;
  max-width: 100%;
  border: ${(p) => p.theme.borders.base};
  border-radius: ${(p) => p.theme.radii.base};
  border-color: ${(p) => p.theme.colors.gray400};
  outline: none;
  background-color: ${(p) => p.theme.colors.light};
  color: ${(p) => p.theme.colors.gray700};
  line-height: ${(p) => p.theme.lineHeight};
  cursor: pointer;
  appearance: none;

  &:hover {
    border-color: ${(p) => p.theme.colors.primary};
  }

  ${(p) =>
    p.focused &&
    css`
      border-color: ${p.theme.colors.primary};
    `}

    ${(p) =>
      p.invalid &&
      css`
        border-color: ${p.theme.colors.danger} !important;
      `}

  ${(p) =>
    p.disabled &&
    css`
      border-color: ${p.theme.colors.gray400};
      background-color: ${p.theme.colors.gray300};
      cursor: not-allowed;

      &:hover {
        border-color: ${p.theme.colors.gray400};
      }

      input {
        color: ${p.theme.colors.gray500};
      }

      ${StyledSelectedOption} {
        border-color: ${p.theme.colors.gray400};
        background-color: ${p.theme.colors.gray200};
        color: ${p.theme.colors.gray400};
        pointer-events: none;
      }
    `}

  input {
    flex: 1;
    width: 100%;
    overflow: hidden;
    border: none;
    outline: none;
    background-color: transparent;
    text-overflow: ellipsis;
    pointer-events: none;

    &::placeholder {
      color: ${(p) => p.theme.colors.gray400};
    }
  }

  ${({ size = 'md', multiple, theme: { paddings, heights, fontSizes } }) => {
    const inputSizeStyles = {
      sm: css`
        min-height: ${heights.sm};
        padding: ${multiple ? '0.5px' : '1px'} ${paddings.xs};

        input {
          font-size: ${fontSizes.sm};
        }
      `,
      md: css`
        min-height: ${heights.base};
        padding: ${multiple ? '2px' : '4px'} ${paddings.sm};

        input {
          font-size: ${fontSizes.base};
        }
      `,
      lg: css`
        min-height: ${heights.lg};
        padding: ${multiple ? '3px' : '6px'} ${paddings.md};

        input {
          font-size: ${fontSizes.lg};
        }
      `,
    };

    return inputSizeStyles[size];
  }};

  ${(p) => p.theme.transition /* sc-declaration */};
`;

const spin = keyframes`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`;

export const Loading = styled.div`
  flex: none;
  width: 24px;
  height: 24px;
  margin: 4px;
  border: ${(p) => p.theme.borders.base};
  border-radius: 50%;
  border-top-color: transparent;
  border-right-color: transparent;
  animation: ${spin} 0.7s linear infinite;
`;
