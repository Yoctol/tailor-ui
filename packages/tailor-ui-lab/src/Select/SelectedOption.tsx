import React, { FunctionComponent } from 'react';
import { MdClose } from 'react-icons/md';

import { Icon } from 'tailor-ui';

import { StyledSelectedOption } from './styles';
import { getDataTestId } from './utils';

export interface SelectedOption {
  index: number;
  'data-testid'?: string;
}

const SelectedOption: FunctionComponent<SelectedOption> = ({
  children,
  index,
  ...props
}) => {
  return (
    <StyledSelectedOption {...getDataTestId(props, `selected-option-${index}`)}>
      {children}
      <Icon
        type={MdClose}
        size="16"
        cursor="pointer"
        ml="1"
        {...props}
        {...getDataTestId(props, `selected-option-${index}-clear-icon`)}
      />
    </StyledSelectedOption>
  );
};

export default SelectedOption;
