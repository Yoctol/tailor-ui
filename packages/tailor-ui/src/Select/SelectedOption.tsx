import React, { FC } from 'react';
import { MdClose } from 'react-icons/md';

import { Icon } from '../Icon';

import { StyledSelectedOption } from './styles';
import { getDataTestId } from './utils';

export interface SelectedOption {
  index: number;
  size: 'sm' | 'md' | 'lg';
  'data-testid'?: string;
}

const SelectedOption: FC<SelectedOption> = ({ children, index, ...props }) => {
  return (
    <StyledSelectedOption
      {...props}
      {...getDataTestId(props, `selected-option-${index}`)}
    >
      {children}
      <Icon
        type={MdClose}
        size="16"
        cursor="pointer"
        ml="1"
        {...getDataTestId(props, `selected-option-${index}-clear-icon`)}
      />
    </StyledSelectedOption>
  );
};

export default SelectedOption;
