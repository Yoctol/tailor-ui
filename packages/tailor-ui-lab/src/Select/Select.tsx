import Downshift from 'downshift';
import React, { FunctionComponent, ReactNode, useRef, useState } from 'react';

import { Flex, Popover, Position, useFormField } from 'tailor-ui';

import ClearIcon from './ClearIcon';
import MultiDownshift from './MultiDownshift';
import SelectArrow from './SelectArrow';
import SelectInput from './SelectInput';
import SelectOptions, { ObjectOption, Option } from './SelectOptions';
import SelectedOption from './SelectedOption';
import { Loading, SelectWrapper, StyledSelect } from './styles';
import { getDataTestId, itemToString } from './utils';

interface SelectProps {
  id?: string;
  name?: string;
  className?: string;
  width?: string | number;
  size?: 'sm' | 'md' | 'lg';
  creatable?: boolean;
  clearable?: boolean;
  disabled?: boolean;
  loading?: boolean;
  multiple?: boolean;
  searchable?: boolean;
  options: Option[];
  value?: Option;
  defaultValue?: Option;
  placeholder?: string;
  menu?: ReactNode;
  itemSize?: number;
  optionsMaxHeight?: number;
  onChange?: (option: Option) => void;
  noOptionsMessage?: () => ReactNode;
  formatCreateLabel?: (labelInfo: {
    value: string;
    active: boolean;
    hovered: boolean;
  }) => ReactNode;
  isValidNewOption?: (value: string) => boolean;
  onCreateOption?: (value: string) => void;
  'data-testid'?: string;
}

const Select: FunctionComponent<SelectProps> = ({
  id,
  width = 240,
  size = 'md',
  creatable = false,
  clearable = false,
  disabled = false,
  loading = false,
  multiple = false,
  searchable = false,
  options,
  value,
  defaultValue,
  placeholder = '',
  menu,
  itemSize = 36,
  optionsMaxHeight = 180,
  onChange,
  noOptionsMessage,
  formatCreateLabel,
  isValidNewOption,
  onCreateOption,
  ...props
}) => {
  const [invalid, labelId, setValue] = useFormField({
    id,
    value,
    defaultValue,
  });
  const inputRef = useRef<HTMLInputElement>(null);
  const [visible, setVisible] = useState(false);
  const [inputValue, setInputValue] = useState('');

  const handleChange = (selection: Option) => {
    if (inputRef.current && !multiple) {
      inputRef.current.blur();
    }

    const isCreate =
      selection && (selection as ObjectOption).label === 'CREATE_OPTION';

    if (isCreate && onCreateOption) {
      onCreateOption((selection as ObjectOption).value);
    }

    if (!isCreate && onChange) {
      onChange(selection);
      setValue(selection);
    }

    if (!multiple || isCreate) {
      setVisible(false);
    } else {
      setInputValue('');
    }
  };

  const defaultHighlightedIndex = multiple
    ? null
    : options.findIndex(
        option => itemToString(option) === itemToString(value || defaultValue)
      ) || null;

  const RenderComponent = multiple ? MultiDownshift : Downshift;

  return (
    <RenderComponent
      id={labelId}
      selectedItem={value}
      isOpen={visible}
      initialSelectedItem={defaultValue}
      defaultHighlightedIndex={defaultHighlightedIndex}
      stateReducer={(state, changes) => {
        // if the user is opening the menu, then let's make sure
        // that the highlighted index is set to the selected index
        if (
          typeof changes === 'object' &&
          changes.type === Downshift.stateChangeTypes.mouseUp
        ) {
          return {};
        }

        return changes;
      }}
      itemToString={itemToString}
      inputValue={inputValue}
      onInputValueChange={newInputValue => {
        if (!multiple && newInputValue !== 'CREATE_OPTION') {
          setInputValue(newInputValue);
        }
      }}
      onChange={handleChange}
      {...{
        scrollIntoView: () => {},
      } as any}
    >
      {({
        id: downshiftId,

        getRootProps,
        getMenuProps,
        getItemProps,
        getInputProps,
        highlightedIndex,
        selectedItem,
        clearSelection,

        selectedItems = [],
        getRemoveButtonProps,
        removeItem,
      }: any) => (
        <SelectWrapper
          {...getRootProps({
            style: { width },
          })}
        >
          <Popover
            visible={visible}
            position={Position.BOTTOM_LEFT}
            minWidth={width}
            onVisibleChange={newVisible => {
              if (disabled || loading) {
                return;
              }

              if (!newVisible && inputRef.current) {
                inputRef.current.blur();
              }

              setVisible(newVisible);
            }}
            p="0"
            content={
              <SelectOptions
                visible={visible}
                creatable={creatable}
                searchable={searchable}
                multiple={multiple}
                itemSize={itemSize}
                optionsMaxHeight={optionsMaxHeight}
                getMenuProps={getMenuProps}
                getItemProps={getItemProps}
                options={options}
                inputValue={inputValue}
                menu={menu}
                highlightedIndex={highlightedIndex}
                selectedItem={selectedItem}
                selectedItems={selectedItems}
                noOptionsMessage={noOptionsMessage}
                formatCreateLabel={formatCreateLabel}
                isValidNewOption={isValidNewOption}
                data-testid={props['data-testid']}
              />
            }
          >
            <StyledSelect
              id={downshiftId}
              {...getDataTestId(props)}
              invalid={invalid}
              size={size}
              focused={visible}
              disabled={disabled || loading}
              onClick={() => {
                if (!visible) {
                  if (searchable || creatable || multiple) {
                    setInputValue('');
                  }
                  if (inputRef.current) {
                    inputRef.current.focus();
                  }
                }
              }}
            >
              <Flex
                flex="auto"
                flexWrap="wrap"
                alignItems="center"
                overflow="hidden"
              >
                {multiple &&
                  selectedItems.map((item: Option) => (
                    <SelectedOption
                      key={itemToString(item)}
                      {...getRemoveButtonProps({
                        item,
                        onClick: () => {
                          if (inputRef.current) {
                            inputRef.current.focus();
                          }
                        },
                      })}
                    >
                      <span>{itemToString(item)}</span>
                    </SelectedOption>
                  ))}
                <SelectInput
                  ref={inputRef}
                  visible={visible}
                  searchable={searchable}
                  creatable={creatable}
                  multiple={multiple}
                  inputValue={inputValue}
                  placeholder={placeholder}
                  selectedItem={selectedItem}
                  selectedItems={selectedItems}
                  removeItem={removeItem}
                  onChange={event => setInputValue(event.currentTarget.value)}
                  getInputProps={getInputProps}
                  data-testid={props['data-testid']}
                />
              </Flex>
              {clearable && selectedItem && (
                <ClearIcon clearSelection={clearSelection} />
              )}
              {loading ? (
                <Loading title="loading" />
              ) : (
                <SelectArrow on={visible} />
              )}
            </StyledSelect>
          </Popover>
        </SelectWrapper>
      )}
    </RenderComponent>
  );
};

export { Select };
