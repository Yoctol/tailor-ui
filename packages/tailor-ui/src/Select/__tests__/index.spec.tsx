import React from 'react';

import { render } from 'test/test-utils';

import Select from '../Select';

describe('Select', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Select
        value={{ label: 'Item 1', value: '1' }}
        options={[
          {
            label: 'Item 1',
            value: '1',
          },
          {
            label: 'Item 2',
            value: '2',
          },
        ]}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should support data-testid', () => {
    const { getByTestId } = render(
      <Select
        value={{ label: 'Item 1', value: '1' }}
        options={[
          {
            label: 'Item 1',
            value: '1',
          },
          {
            label: 'Item 2',
            value: '2',
          },
        ]}
        data-testid="my-select"
      />
    );

    expect(getByTestId('my-select')).toBeInTheDocument();
  });

  it('should support data-testid for options', () => {
    const { getByTestId } = render(
      <Select
        value={{ label: 'Item 1', value: '1' }}
        options={[
          {
            label: 'Item 1',
            value: '1',
            'data-testid': 'my-option-1',
          },
          {
            label: 'Item 2',
            value: '2',
            'data-testid': 'my-option-2',
          },
        ]}
      />
    );

    expect(getByTestId('my-option-1')).toBeInTheDocument();
    expect(getByTestId('my-option-1')).toBeInTheDocument();
  });
});
