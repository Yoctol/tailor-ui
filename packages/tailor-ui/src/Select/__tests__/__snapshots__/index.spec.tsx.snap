// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
.c0 .yoctol-select__control {
  min-width: 150px;
  border-color: #e0e0e0;
  background-color: #ffffff;
  box-shadow: none;
}

.c0 .yoctol-select__control:hover {
  border-color: #423b63;
}

.c0 .yoctol-select__control.yoctol-select__control--is-focused {
  border-color: #191539;
}

.c0 .yoctol-select__control.yoctol-select__control-is-disabled .yoctol-select__indicators::before {
  opacity: 0.5;
}

.c0 .yoctol-select__menu {
  margin-top: 2px;
}

.c0 .yoctol-select__menu .yoctol-select__menu-list {
  padding-top: 0;
  padding-bottom: 0;
}

.c0 .yoctol-select__menu .yoctol-select__menu-list .yoctol-select__option {
  background-color: #ffffff;
  color: #373737;
}

.c0 .yoctol-select__menu .yoctol-select__menu-list .yoctol-select__option:not(:last-child) {
  border-bottom: 1px solid #e0e0e0;
}

.c0 .yoctol-select__menu .yoctol-select__menu-list .yoctol-select__option.yoctol-select__option--is-focused {
  background-color: #191539;
  color: #ffffff;
}

<div
  class="css-10nd86i c0"
>
  <div
    class="css-vj8t7z yoctol-select__control"
  >
    <div
      class="css-1hwfws3 yoctol-select__value-container yoctol-select__value-container--has-value"
    >
      <div
        class="css-xp4uvy yoctol-select__single-value"
      >
        Item 1
      </div>
      <input
        class="css-14uuagi"
        id="react-select-2-input"
        readonly=""
        tabindex="0"
        value=""
      />
    </div>
    <div
      class="css-1wy0on6 yoctol-select__indicators"
    >
      <span
        class="css-d8oujb yoctol-select__indicator-separator"
      />
      <div
        aria-hidden="true"
        class="css-1ep9fjw yoctol-select__indicator yoctol-select__dropdown-indicator"
      >
        <svg
          aria-hidden="true"
          class="css-19bqh2r"
          focusable="false"
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
